Unity/opentok/holo/holo/Capturer.mm

```diff
@@ -0,0 +1,373 @@
+    [condition lock];
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^(void){
+        [_capturer stopWithCompletionHandler:^{
+            [condition signal];
+        }];
+    });
+    [condition wait];
+    return 0;
+}
+
+- (BOOL)isCaptureStarted {
+    return _capturing;
+}
+
+- (int32_t)captureSettings:(OTVideoFormat*)videoFormat {
+    videoFormat.pixelFormat = OTPixelFormatNV12;
+    videoFormat.imageWidth = _captureWidth;
+    videoFormat.imageHeight = _captureHeight;
+    return 0;
+}
+
+- (void)updateCaptureFormatWithWidth:(uint32_t)width height:(uint32_t)height {
+    _captureWidth = width;
+    _captureHeight = height;
+    [_videoFrame setFormat:[OTVideoFormat
+                            videoFormatNV12WithWidth:_captureWidth
+                            height:_captureHeight]];
+}
+
+-(void)callDelegateOnError:(OTError*)error captureError:(NSError *)captureError {
+    _captureErrorCode = (enum OTCapturerErrorCode)error.code;
+}
+
+-(enum OTCapturerErrorCode)captureError {
+    return _captureErrorCode;
+}
+
+- (void)capturer:(nonnull RTC_OBJC_TYPE(RTCVideoCapturer) *)capturer didCaptureVideoFrame:(nonnull RTC_OBJC_TYPE(RTCVideoFrame) *)videoFrame {
+//        NSLog(@"[holo]: Capturer %p capturer New frame captured (width %d and height %d)", self, videoFrame.width, videoFrame.height);
+
+    if ([videoFrame.buffer isKindOfClass:[RTC_OBJC_TYPE(RTCCVPixelBuffer) class]]) {
+        CVPixelBufferRef buffer = ((RTC_OBJC_TYPE(RTCCVPixelBuffer) *)videoFrame.buffer).pixelBuffer;
+        OTVideoOrientation orientation = OTVideoOrientationUp;
+        switch ([videoFrame rotation]) {
+            case RTCVideoRotation_0:
+                orientation = OTVideoOrientationUp;
+                break;
+            case RTCVideoRotation_90:
+                orientation = OTVideoOrientationLeft;
+                break;
+            case RTCVideoRotation_180:
+                orientation = OTVideoOrientationDown;
+                break;
+            case RTCVideoRotation_270:
+                orientation = OTVideoOrientationRight;
+                break;
+
+            default:
+                orientation = OTVideoOrientationUp;
+                break;
+        }
+        NSData *data = nil;
+        if ([videoFrame isAugmented]) {
+            data = [NSData dataWithBytes:[videoFrame augmentingData] length:[videoFrame augmentingDataSize]];
+//            NSLog(@"[holo]: Capturer %p capturer augmented data size is %lu", self, static_cast<size_t>(data.length));
+        }
+        // TODO: Review the way we build the timestamp we provide through CMTimeMake (not sure is correct).
+        [_videoCaptureConsumer consumeImageBuffer:buffer
+                                      orientation:orientation
+                                        timestamp: CMTimeMake([videoFrame timeStampNs], CMTimeScale(NSEC_PER_SEC))
+                                         metadata:data];
+
+    }
+}
+
+- (void)setupListenerBlocks {
```

Comment:
i dont think this function is used.

Reply:
I removed it. Not sure why it was there. Thanks.